### ISSUE TITLE
Add CONTRIBUTING.md

### ISSUE THREAD

**Issue by ryanbmilbourne** [spectcl/spectcl#89]
We forgot to add contribution guidelines.

**Comment by ryanbmilbourne** [comment]
@gcochard: Aside from atomic commits using imperative language, is there anything else you'd like to add to the contribution guidelines? 

1) Manual edits to changelog, or generate changelog based on commits?  `eslint` utilizes a useful tagging system to generate changelog, but it would require additions to the build process to generate and commit said changelog from the commits.
2) version rolling.  Should we require pull requests to roll the relevant semver level, or put it on a maintainer to roll versions?

**Comment by gcochard** [comment]
1) Let's go with manual edits to the changelog until such time as we get tired of it.
2) Versioning should *always* be up to the maintainer.

**Comment by gcochard** [comment]
One thing I realized is that with our protected master branch, we can't push directly, so we'll have to create a separate PR for a version bump. While this is somewhat annoying, I don't see any way around it without relinquishing control over the version. It should stay in the hands of the maintainer, just like publishing new releases.

Once the PR is merged, we can create a tag for the version and npm publish it.

**Comment by ryanbmilbourne** [comment]
:+1.  I'll formalize something and make a PR.

**Comment by ryanbmilbourne** [comment]
@gcochard: I have a first draft available [here](https://github.com/ryanbmilbourne/spectcl/blob/add-contributing/CONTRIBUTING.md).  Feel free to make any comments/suggestions.  We might need to flesh out some of the sections a bit more; let me know what you think.
